Use kinder to replace minikube

namnx228 opened the pull request.

Jaakko-Os commented:
Commit message could be rewritten. Now it spreads to multiple lines and several empty lines.

maelk commented on 01_install_requirements.sh:
```diff
@@ -1,23 +1,60 @@
-    sudo mv minikube /usr/local/bin/.
-fi
+# Create Go directory
+mkdir -p ${HOME}/go/{bin,src,pkg/mod} 
```
rather use the GOPATH variable, moving the go env call if needed

maelk commented on vm-setup/libvirt-package-playbook.yml:
```diff
@@ -0,0 +1,11 @@
+---
```
remove, unused

maelk commented on vm-setup/vars/package_installation.yml:
```diff
@@ -0,0 +1,95 @@
+packages_installation:
```
the file name should maybe be packageS too ?

maelk commented on vm-setup/roles/packages_installation/tasks/main.yml:
```diff
@@ -20,17 +20,17 @@
           when: ansible_lsb.major_release|int == 8
           package:
-             name: "{{ install.centos.package + install.centos.libvirt.rhel7 }}"
+             name: "{{ package.centos.package + package.centos.libvirt.rhel7 }}"
```
this needs to be changed to rhel8

maelk commented on vm-setup/install-package-playbook.yml:
```diff
@@ -4,12 +4,12 @@
   gather_facts: true
   vars_files:
-    - vars/input.yml
+    - vars/package_installation.yml
```
packages_... rather ?

maelk commented on README.md:
```diff
@@ -32,7 +32,7 @@ The `Makefile` runs a series of scripts, described here:
   were bare metal hosts.
 
-* `03_launch_mgmt_cluster.sh` - Launch a management cluster using `minikube` and
+* `03_launch_mgmt_cluster.sh` - Launch a management cluster using `kinder` and
```
minikube or kinder

maelk committed:
add option to run local operators

Instead of using 'make deploy' for Baremetal Operator and CAPI
provider baremetal, add an option to use 'make run' and log the
output.

maelk commented on 03_launch_mgmt_cluster.sh:
```diff
@@ -59,6 +59,23 @@ function launch_minikube() {
+  kinder="${GOPATH}/bin/kinder"
+  if [[ $(sudo $kinder get clusters) == "" ]]; then
+    sudo $kinder create cluster
+    sudo $kinder do kubeadm-init
```
the cluster won't have an interface in the provisioning network, that is problematic if for example we move ironic inside the cluster

maelk committed:
Add retries in verification script

Avoid failing without retrying to wait for the dev env
to initialize properly. Add verify target in Makefile all

nordixinfra commented:
Can one of the admins verify this patch?

maelk committed:
add doc and test for running operators locally

- add code documentation for environment variables
- add test to check that the operators are running locally
- add user example

maelk committed:
Add a test script

the script provisions the machines and tests the setup. It will
also deprovision and test

maelk commented on lib/common.sh:
```diff
@@ -33,6 +34,13 @@ INT_IF=${INT_IF:-}
 ROOT_DISK_NAME=${ROOT_DISK_NAME-"/dev/sda"}
 #Container runtime
+
+K8S=${K8S:-"minikube"}
```
Please add this in config_example.sh

namnx228 replied:
Done

maelk commented on vm-setup/roles/libvirt/tasks/install_package_tasks.yml:
```diff
@@ -0,0 +1,27 @@
+# Install the packages required for our desired libvirt environment.
```
this should not be in this PR

russellb committed:
Merge pull request #43 from Nordix/ci-mael

add option to run local operators

maelk committed:
Refactor 04_verify.sh and 05_test.sh bash code

maelk committed:
Use containerized vbmc and add sushy-tool for redfish interface

maelk committed:
add Dockerfiles for vbmc and redfish emulator containers

Co-authored-by: Nam Xuan Nguyen <nam.xuan.nguyen@est.tech>

russellb committed:
Merge pull request #45 from Nordix/verif-mael

Add retry loop for deployment verification and test script

Commit:
Move all download code into 01_install_requirement.sh

New ansible playbook to download packages 


Install redhat-lsb-core  using yum to make  `ansible_lsb` available

Fix typos


Update package playbook with two level of centos: rhel7 and rhel8


Remove some packages that are  not in RHEL8

russellb committed:
README: Replace content with metal3.io link.

Most of this content has moved to http://metal3.io/try-it.html, so
drop it from this file to avoid duplicating it.

russellb committed:
Merge pull request #49 from russellb/metal3-io-try-it

README: Replace content with metal3.io link.

namnx228 committed:
Fix errors on RHEL 8 and Ubuntu

derekhiggins committed:
Add support for UEFI VMs

derekhiggins committed:
Switch VM machine type to q35

When needing UEFI boot in future this will be needed
as i440fx isn't supported by the OVMF provided with
RHEL/CentOS.

From https://bugzilla.redhat.com/show_bug.cgi?id=1308678#c10
"there will be no support for i440fx machine types"

Commit:
Merge pull request #52 from derekhiggins/uefi

Uefi

russellb committed:
Merge pull request #37 from Nordix/improvement/namnx228-download-1-place

Move all download code into 01_install_requirement.sh

Commit:
Fix epel install on RHEL

This currently doesn't work on either RHEL version

Commit:
Remove unused lolcat/figlet packages

These were removed from dev-scripts in openshift-metal3/dev-scripts/pull/595
and they aren't needed in metal3-dev-env

Commit:
Move packages_installation global vars to role defaults

The normal pattern is to keep the per-role defaults in the role
directory AFAIK, this can still be overridden via vars files if needed.

Commit:
Install pip via packages_installation

This can be handled per-version more easily in the ansible defaults,
since the package name differs betwen el7 and el8

Commit:
Add ansible-2-for-rhel-8-x86_64-rpms repo for RHEL8

Otherwise the install of ansible will fail

Commit:
Set alternatives on RHEL8 for python3

This means we can install tripleo-repos, which now seems to work
OK on RHEL8 \o/

Commit:
Remove EPEL for RHEL8

AFAICS this is not actually needed, because we get the ansible/golang
things needed from the other enabled repos.

Commit:
Use absolute path for tripleo-repos

This ends up installed in /usr/local/bin on RHEL8, which isn't in
the restricted $PATH when running via sudo - this seems safer than
preserving some or all of the parent shell environment.

Commit:
Install virtualbmc from RPM on RHEL8

This is now available via a package on RHEL8 with the repos configured
via tripleo-repos

Commit:
Install deprecated network-scripts package on RHEL8

Until we replace the legacy network scripts e.g ifcfg-provisioning
and related ifup/ifdown calls with nmcli, we'll need this

Commit:
Add crudini package on RHEL8

Commit:
Install tripleo-repos via package

If we use dnf --repofrompath we can install this from the relevant
RDO repo instead of from source, which will help resolve dependency
issues such as https://github.com/openshift-metal3/dev-scripts/issues/805

Commit:
RHEL8 distro setup shouldn't depend on epel

Fix the conditional logic here so it's independent of any
epel repo setup (which shouldn't be required for RHEL8 but
could potentially be configured by a user which would break
the current logic).

Commit:
Install python3-openstackclient on RHEL8

maelk committed:
Add more verbose messages for test failures

russellb committed:
Ignore failure from "minikube delete".

When running "make clean" on a system where the minikube VM isn't
running, "minikube delete" will return non-zero and the rest of the
cleanup process won't run.  Instead, ignore any non-zero return from
"minikube clean" to make sure we run the other clean steps.

maelk committed:
move minikube initialization to step 01

this is to optimize CI runs where 01 requirements are baked in

russellb committed:
Merge pull request #54 from hardys/package_cleanups

Cleanups and fixes for RHEL8

russellb committed:
Ensure dnf gets installed on an existing centos 7 host.

dnf was previously only installed in the case that EPEL was not yet
enabled.  Split the dnf install out into its own command instead.

Commit:
Merge pull request #55 from Nordix/verbose-test-mael

Add more verbose messages for test failures

maelk committed:
Fix shellcheck errors

russellb committed:
Merge pull request #57 from Nordix/lint-fix-mael

Fix shellcheck errors

Commit:
Ensure python3 is installed on RHEL8

Turns out this is not available when you use a minimal install
so lets ensure it's installed before the call to alternatives,
which fails when /usr/bin/python3 does not exist.

Reported via https://github.com/openshift-metal3/dev-scripts/issues/811

Closes: openshift-metal3/dev-scripts/issues/811

russellb committed:
Merge pull request #58 from hardys/dev_scripts_811

Ensure python3 is installed on RHEL8

russellb committed:
Merge pull request #56 from Nordix/centos-fix-mael

move minikube initialization to step 01

namnx228 committed:
Use kinder to replace minikube


Change file 01_requirement to 01_prepare

namnx228 committed:
Remove some redundancy lines

maelk committed:
Fix libvirt group issue for Minikube

russellb committed:
Merge pull request #59 from Nordix/minikube-fix-mael

Fix libvirt group issue for Minikube

maelk committed:
Prevent missing group issue with Minikube in CI

When starting minikube in the same script as we add the user to
libvirt group, it fails. This runs the Minikube commands in a
new login shell

russellb committed:
Merge pull request #48 from Nordix/vbmc-mael

Add redfish emulator and containerize VBMC

russellb committed:
Merge pull request #60 from Nordix/minikube-fix-mael

Prevent missing group issue with Minikube in CI

kashifest committed:
Add a switch to use redfish as BMC

Commit:
Reinstate virtualbmc service teardown tasks

These were removed in #48 which is fine for a greenfield CI environment
but leads to unexpected behavior in developer test environments, since
the old systemd service is not stopped before starting the new containers.

russellb committed:
Merge pull request #63 from hardys/vbmc_fixes

Reinstate virtualbmc service teardown tasks

Commit:
source lib/common.sh in 01_prepare_host.sh

Currently VBMC_IMAGE and SUSHY_TOOLS_IMAGE are undefined because
they're defined in lib/common.sh.

Also I noticed that the pushd uses a relative path which won't work
unless the calling shell is in the metal3-dev-env dir, and we're
not logging the output from this script.

russellb committed:
Merge pull request #64 from hardys/vbmc_fixes_2

source lib/common.sh in 01_prepare_host.sh

russellb committed:
Use vbmc and sushy-tools images from quay.io.

Instead of building these locally, pull pre-built images from quay.io.
These images are set to automatically rebuild any time metal3-dev-env
changes.

russellb committed:
Merge pull request #65 from russellb/quay-images

Use vbmc and sushy-tools images from quay.io.

namnx228 committed:
Update access right of 01_preapre_host.sh

maelk commented:
add to whitelist

Commit:
Add entries to cloud init

- Install and setup keepalived
- Pass config file to keepalived
- Install required binaries
- run kubeadm init with VIP
- configure kubectl to access api-server on VIP

maelk commented:
we will not take this further for now
